packages/core/shared/parse-complex-response.ts

```diff
@@ -15,20 +15,12 @@ type PrefixMap = {
   data: JSONValue[];
 };
 
-function initializeMessage({
-  generateId,
-  ...rest
-}: {
-  generateId: () => string;
-  content: string;
-  createdAt: Date;
-  annotations?: JSONValue[];
-}): Message {
-  return {
-    id: generateId(),
-    role: 'assistant',
-    ...rest,
-  };
+function assignAnnotationsToMessage<T extends Message | null | undefined>(
+  message: T,
+  annotations: JSONValue[] | undefined,
+): T {
+  if (!(message && annotations && annotations.length)) return message;
```

Comment:
small nit, I prefer the expanded version

```suggestion
  if (!message || !annotations || !annotations.length) return message;
```